### ISSUE TITLE
Error resolving dependency com.github.everit-org.json-schema:org.everit.json.schema:1.12.1

### ISSUE THREAD

**Issue by lislei** [CycloneDX/cyclonedx-core-java#48]
When using the [cyclonedx-gradle-plugin](https://github.com/CycloneDX/cyclonedx-gradle-plugin) version 1.2.0 build produces this error:
```
> Could not resolve all artifacts for configuration ':classpath'.
   > Could not find com.github.everit-org.json-schema:org.everit.json.schema:1.12.1.
     Searched in the following locations:
       - file:/C:/Users/lislei/.m2/repository/com/github/everit-org/json-schema/org.everit.json.schema/1.12.1/org.everit.json.schema-1.12.1.pom
       - https://nexus.statkart.no/repository/public/com/github/everit-org/json-schema/org.everit.json.schema/1.12.1/org.everit.json.schema-1.12.1.pom
     Required by:
         project : > org.cyclonedx:cyclonedx-gradle-plugin:1.2.0 > org.cyclonedx:cyclonedx-core-java:3.0.5
```

The problem is that a dependency is not publishet in maven-central. 

The easiest solution would be to have the dependency published to maven central.
If not, is it feasable to choose a different implementation for json parsing?

**Comment by stevespringett** [comment]
Often times, local repositories (e.g. artifactory, nexus, etc) are misconfigured and assume Maven Central for all Java libraries. This is obviously not the case. 

There are no intentions of choosing another library. The chosen library is ideal for CycloneDX Java Core. The author moved from Maven Central to Jitpack due to the increased security and transparency Jitpack provides over Central.

I would recommend working with the build engineering team to correct any Maven or repository settings so you're able to properly resolve it.

Another possibility would be to shade the library into Java Core, but this will need research into what kinds of downstream effects it may have. So, this is not something that can be done in the short term.

**Comment by lislei** [comment]
Sorry for the late update. As you figured I missed to mention that we're using a local, company wide, repository.
It is configured to mirror the most common repositories needed.

As a workaround we added the [jar file](https://github.com/everit-org/json-schema/releases/download/1.12.1/everit-json-schema-1.12.1.jar) to our company wide repository. Had to make some modifications to the pom as the group name differs from the source code one on github. 

For downstream projects, it looks like the documentation is incomplete at times. In my case of setting up [cyclonedx-gradle-plugin](https://github.com/CycloneDX/cyclonedx-gradle-plugin) it instructs to use Maven Central. 

For a long term solution I would vote for shading `everit-json-schema` into the jar. This eliminated the confusion of having to retrieve dependencies across severl repos. 
Im not an expert on Maven, but one have to delcare the transitive dependencies somehow:
```
     |    \--- com.github.everit-org.json-schema:org.everit.json.schema:1.12.1
     |         +--- org.json:json:20190722
     |         +--- commons-validator:commons-validator:1.6
     |         |    +--- commons-digester:commons-digester:1.8.1
     |         |    +--- commons-logging:commons-logging:1.2
     |         |    \--- commons-collections:commons-collections:3.2.2
     |         +--- com.damnhandy:handy-uri-templates:2.1.8
     |         |    \--- joda-time:joda-time:2.10.2 -> 2.10.5
     |         \--- com.google.re2j:re2j:1.3
```

**Comment by stevespringett** [comment]
@mikejhill Thanks for uncovering that fact and for pointing out the artifact from Central is from the same author.

**Comment by stevespringett** [comment]
This will be resolved in CycloneDX Java Core v4.0.0 which is currently under active development.

**Comment by stevespringett** [comment]
CycloneDX Java Core v4.0.0 has been released to Central. Closing.
